test/scripts/e2e-kind.sh

```diff
@@ -47,26 +47,37 @@ should prevent Ingress creation if more than 1 IngressClass marked as default
 
 SKIPPED_TESTS=$(echo "${SKIPPED_TESTS}" | sed -e '/^\($\|#\)/d' -e 's/ /\\s/g' | tr '\n' '|' | sed -e 's/|$//')
 
-GINKGO_ARGS="--num-nodes=3 --ginkgo.skip=${SKIPPED_TESTS} --disable-log-dump=false --report-dir=${E2E_REPORT_DIR} --report-prefix=${E2E_REPORT_PREFIX}"
+# if we set PARALLEL=true, skip serial test
+if [ "${PARALLEL:-true}" = "true" ]; then
+  export GINKGO_PARALLEL=y
+  export GINKGO_PARALLEL_NODES=10
+  SKIPPED_TESTS="${SKIPPED_TESTS}|\\[Serial\\]"
+fi
 
 case "$SHARD" in
-	shard-n-other)
-		# all tests that don't have P as their sixth letter after the N, and all other tests
-		GINKGO_ARGS="${GINKGO_ARGS} "'--ginkgo.focus=\[sig-network\]\s([Nn](.{6}[^Pp].*|.{0,6}$)|[^Nn].*)'
+	shard-network)
+		FOCUS="\\[sig-network\\]"
 		;;
-	shard-np)
-		# all tests that have P as the sixth letter after the N
-		GINKGO_ARGS="${GINKGO_ARGS} "'--ginkgo.focus=\[sig-network\]\s[Nn].{6}[Pp].*$'
+	shard-conformance)
+		FOCUS="\\[Conformance\\]"
 		;;
 	shard-test)
-		TEST_REGEX_REPR=$(echo ${@:2} | sed 's/ /\\s/g')
-		GINKGO_ARGS="${GINKGO_ARGS} "'--ginkgo.focus=\[sig-network\].*'$TEST_REGEX_REPR'.*'
+		FOCUS=$(echo ${@:2} | sed 's/ /\\s/g')
 		;;
 	*)
 		echo "unknown shard"
 		exit 1
 	;;
 esac
 
-e2e.test ${GINKGO_ARGS}
-
+# setting this env prevents ginkgo e2e from trying to run provider setup
+export KUBERNETES_CONFORMANCE_TEST='y'
+# setting these is required to make RuntimeClass tests work ... :/
+export KUBE_CONTAINER_RUNTIME=remote
+export KUBE_CONTAINER_RUNTIME_ENDPOINT=unix:///run/containerd/containerd.sock
+export KUBE_CONTAINER_RUNTIME_NAME=containerd
+NUM_NODES=2
```

Comment:
What is NUM_NODES represent? The reason I ask is because kind.sh allows updates KIND_NUM_MASTER based on OVN_HYBRID_OVERLAY_ENABLE being true or false and allows KIND_NUM_WORKER to be manually configured.

Reply:
is an option of the e2e framework, I think that some tests use it in a sense that if the cluster does not have more than X nodes I skip the test, honestly I don't know also if some test use it for anything else, just carrying over the same numbers :smile: 
```
  -num-nodes int
    	Number of nodes in the cluster. If the default value of '%!q(int=-1)' is used the number of schedulable nodes is auto-detected. (default -1)
```